stream-chat-swift-docs/docs/guides/getting-started.md

```diff
@@ -1,3 +1,149 @@
+
+### ChatClientConfig
+
+The `ChatClientConfig` object holds properties that the chat client will use to determine certain behaviors. For example, the `apiKey`, which tells the chat client which chat server to communicate with. It also has the `baseURL` property which tells the chat client which region of the world your server is at, which can be useful to reduce overall latency.
+
+### Singleton
+
+To make a `ChatClient` singleton, we extend the `ChatClient` class with a static shared instance of itself.
+
+```swift
+/// 1: Extend the `ChatClient` class.
+extension ChatClient {
+
+    /// 2: Add a `shared` static variable of its own type.
+    static var shared: ChatClient!
+}
+```
+
+In the desired entry point, set the shared instance with the tokenProvider from the Authentication step. In this case, we're doing it in the `AppDelegate`'s `didFinishLaunchingWithOptions` callback.
+
+```swift
+import StreamChat
+import UIKit
+
+@main
+class AppDelegate: UIResponder, UIApplicationDelegate {
+    func application(
+        _ application: UIApplication,
+        didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
+    ) -> Bool {
+        /// 1: The token provider you created in the Authentication section.
+        let token = Token("USER_TOKEN")
+        let tokenProvider = TokenProvider.static(token)
+
+        /// 2: Create a `ChatClientConfig` with the API key.
+        let config = ChatClientConfig(apiKeyString: "YOUR_API_KEY")
+
+        /// 3: Set the shared `ChatClient` instance with the config and the token provider.
+        ChatClient.shared = ChatClient(config: config, tokenProvider: tokenProvider)
+
+        return true
+    }
+}
+```
+
+### Dependency Injection
+
+To create a `ChatClient` instance for dependency injection, just instantiate it normally in a convenient entry point. In this case, we're instantiating it in the SceneDelegate and passing it down to the first view controller. Don't forget to add a `var chatClient: ChatClient!` property to your view controller class.
+
+```swift
+import StreamChat
+import UIKit
+
+class SceneDelegate: UIResponder, UIWindowSceneDelegate {
+    var window: UIWindow?
+
+    func scene(
+        _ scene: UIScene,
+        willConnectTo session: UISceneSession,
+        options connectionOptions: UIScene.ConnectionOptions
+    ) {
+        guard let scene = (scene as? UIWindowScene) else { return }
+        scene.windows.forEach { $0.tintColor = .systemPink }
+
+        /// 1: Get the reference to the initial view controller
+        let viewController = window?.rootViewController as? ViewController
```

Comment:
We should put the simplest ViewController definition up top, so copy/paste/compile works